application/controllers/IdoController.php

```diff
@@ -0,0 +1,101 @@
+<?php
+// Icinga Web 2 Cube Module | (c) 2019 Icinga GmbH | GPLv2
+
+namespace Icinga\Module\Cube\Controllers;
+
+use Icinga\Module\Cube\Web\Controller;
+use Icinga\Module\Cube\Ido\IdoHostStatusCube;
+use Icinga\Module\Cube\Web\ActionLinks;
+use Icinga\Web\Widget\Tabextension\DashboardAction;
+
+class IdoController extends Controller
+{
+    public function addTabs() {
+        return $this->getTabs()
+            ->add('cube/hosts', array(
+                'label' => $this->translate('Hosts'),
+                'url'   => 'cube/hosts' . ($this->params->toString() === '' ? '' : '?' . $this->params->toString())
+            ))
+            ->add('cube/services', array(
+                'label' => $this->translate('Services'),
+                'url'   => 'cube/services' . ($this->params->toString() === '' ? '' : '?' . $this->params->toString())
+            ))
+            ->extend(new DashboardAction());
+    }
+
+    public function buildCube($cube)
+    {
+        // Hint: order matters, we are shifting!
+        $showSettings = $this->params->shift('showSettings');
+
+        $cube->chooseFacts(array_keys($cube->getAvailableFactColumns()));
+        $dimensions = $this->params->shift('dimensions');
+        $wantNull = $this->params->shift('wantNull');
+        $vars = preg_split('/,/', $dimensions, -1, PREG_SPLIT_NO_EMPTY);
+        foreach ($vars as $var) {
+            $cube->addDimensionByName($var);
+            if ($wantNull) {
+                $cube->getDimension($var)->wantNull();
+            }
+        }
+
+        foreach (array('renderLayout', 'showFullscreen', 'showCompact', 'view') as $p) {
+            $this->params->shift($p);
+        }
+
+        foreach ($this->params->toArray() as $param) {
+            $cube->slice($param[0], urldecode($param[1]));
+        }
+
+        $this->view->title = sprintf(
+            $this->translate('Cube: %s'),
+            $cube->getPathLabel()
+        );
+
+        if (count($cube->listDimensions()) > 0) {
+            $this->view->cube = $cube;
+        } else {
+            $showSettings = true;
+        }
+
+        if ($showSettings) {
+            $this->view->form = $this->loadForm('Dimensions')
+                ->setCube($cube)
+                ->handleRequest();
+        } else {
+            $this->setAutorefreshInterval(15);
+        }
+    }
+
+    public function detailsAction()
```

Comment:
Move the `detailsAction()` to the module's base controller class and rely on the protected member variable `$cube`.